#! /bin/sh /usr/share/dpatch/dpatch-run
## 02-edit-wrapper.dpatch by Tiago Saboga <tiagosaboga@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Rewrite shell wrapper:
## DP: - use /bin/sh instead of /bin/bash
## DP: - look for sun's java vm

@DPATCH@
diff -urNad omegat-1.7.3.u2+dfsg~/release/OmegaT omegat-1.7.3.u2+dfsg/release/OmegaT
--- omegat-1.7.3.u2+dfsg~/release/OmegaT	2008-03-28 19:49:26.000000000 +0530
+++ omegat-1.7.3.u2+dfsg/release/OmegaT	2008-11-19 00:13:13.000000000 +0530
@@ -1,3 +1,14 @@
-#!/bin/bash
-java -jar OmegaT.jar $*
+#!/bin/sh
 
+if [ -x /usr/lib/jvm/java-6-openjdk/bin/java ]
+    then JAVABIN=/usr/lib/jvm/java-6-openjdk/bin/java
+elif [ -x /usr/lib/jvm/java-6-sun/bin/java ]
+    then JAVABIN=/usr/lib/jvm/java-6-sun/bin/java
+elif [ -x /usr/lib/jvm/java-1.5.0-sun/bin/java ]
+    then JAVABIN=/usr/lib/jvm/java-1.5.0-sun/bin/java
+else
+    echo "I could not find a suitable java binary."
+    exit 1
+fi
+
+exec $JAVABIN -jar /usr/share/java/OmegaT.jar $*
